.github/workflows/tox.yaml

```diff
@@ -16,7 +16,7 @@ jobs:
         platform:
           - ubuntu-latest
         python-version:
-          - '3.10'
+          - '3.12'
```

Comment:
I had to bump Python version for two reasons:
1. localtuya actually doesn't work with Python < 3.11 because of `from typing import Self`
2. https://github.com/home-assistant/core/pull/114220